cmd/dusk/action.go

```diff
@@ -3,35 +3,51 @@ package main
 import (
 	"bytes"
 	"fmt"
-	"math/rand"
-	"os"
-	"os/signal"
-	"time"
-
 	cfg "github.com/dusk-network/dusk-blockchain/pkg/config"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/message"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/diagnostics"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/logging"
 	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/rpcbus"
-	log "github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"math/rand"
+	"os"
+	"os/signal"
+	"time"
 )
 
-const (
-	// name (wihtout ext) for the config file to look for
-	configFileName = "dusk"
+var (
+	log     *logrus.Entry
+	config  string
+	datadir string
 )
 
-func main() {
+func action(ctx *cli.Context) error {
+
+	// check arguments
+	if arguments := ctx.Args(); len(arguments) > 0 {
+		return fmt.Errorf("failed to read command argument: %q", arguments[0])
+	}
+
+	if datadir = ctx.GlobalString(DataDirFlag.Name); datadir != "" {
+		datadir = "~/./dusk"
+		//TODO:
+	}
+
+	if config = ctx.GlobalString(ConfigFlag.Name); config != "" {
```

Comment:
Doesn't this duplicate  `loadFlags` we already support (--config)?

https://github.com/dusk-network/dusk-blockchain/blob/ffb61f2bf7580bac17c841ce75dbe95f5de77ef3/pkg/config/loader.go#L175

Reply:
good point. However when trying to re-use that method I stumbled in few issues: 1. `loadFlags` its not exported, so its not possible to use it in the main. 2. When making it exported and trying to re-use its logic, I started to face several other issues like: 
`INFO[0000] initializing node...                          app=dusk prefix=main
Dusk node flag redefined: general.network
ERRO[0000] Application panic                             app=dusk error="Dusk node flag redefined: general.network" prefix=main`
3. The code was already like this before, but the config was hardcoded. Instead, I improved a bit by allowing it to be configurable. 

I think we should refactor this flags in another ticket to be raised.

Reply:
raised #367